services/base.js

```diff
@@ -90,9 +90,19 @@ class BaseService {
   static _makeStaticExampleUrl(serviceData) {
     const badgeData = this._makeBadgeData({}, serviceData)
     const color = badgeData.colorscheme || badgeData.colorB
```

Comment:
In terms of refactoring this, there's an `escapeFormat` function https://github.com/badges/shields/blob/bedba47d777cb3b27467fa8b020ac4c5a0105fdd/lib/path-helpers.js#L3-L17 which handles more cases, and heh, another one
https://github.com/badges/shields/blob/bedba47d777cb3b27467fa8b020ac4c5a0105fdd/frontend/lib/badge-url.js#L15-L17

All three of these should be consolidated.

There's also a function which does the same thing as `_makeStaticExampleUrlFromTextAndColor`

https://github.com/badges/shields/blob/bedba47d777cb3b27467fa8b020ac4c5a0105fdd/frontend/lib/badge-url.js#L19-L22

which could be moved from `frontend/lib/badge-url.js` into `lib/path-helpers.js`. That way it can be used in the server, too.

Reply:
At a minimum, instead of patching this function, could you please move [this function](https://github.com/badges/shields/blob/bedba47d777cb3b27467fa8b020ac4c5a0105fdd/frontend/lib/badge-url.js#L19-L22) to `lib/path-helpers.js` and call it here?

Reply:
In the interest of getting this in, let's let stand the change required for `'f-droid'`, and handle the cleanup in a separate issue.